dll/win32/kernel32/client/proc.c

```diff
@@ -4377,6 +4486,176 @@ CreateProcessInternalW(IN HANDLE hUserToken,
         }
     }
 
+    // winsock inheritance handling code
+    if (bInheritHandles) 
+    {
+    	HANDLE PROCESS_HANDLE;
+	    NTSTATUS STATUS = 0x0;
+	    ULONG returnLength = 0;
+
+	    DPRINT("(PID %lx) CreateProcessInternalW: bIneritHandles true, System handle scan for sockets starting\n", GetCurrentProcessId());			
+	    PSYSTEM_HANDLE_INFORMATION handleTableInformation = (PSYSTEM_HANDLE_INFORMATION)HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, SystemHandleInformationSize);  
+	    STATUS = NtQuerySystemInformation(SystemHandleInformation, handleTableInformation, SystemHandleInformationSize, &returnLength);
+	    if(!NT_SUCCESS(STATUS) && (STATUS != STATUS_INFO_LENGTH_MISMATCH))
+	    {
+		    DPRINT1("CreateProcessInternalW: Call to NtQuerySystemInformation failed Status: %ln\n", &STATUS);
+	    }
+	    if(STATUS == STATUS_INFO_LENGTH_MISMATCH)
+	    {
+		    DPRINT1("CreateProcessInternalW: Call to NtQuerySystemInformation failed (STATUS_INFO_LENGTH_MISMATCH) returnLength: %lu\n", returnLength);
+	    }
+	    if(NT_SUCCESS(STATUS)) // we got our access to system handle table so continue..
+	    {
+		    PROCESS_HANDLE = OpenProcess(PROCESS_DUP_HANDLE, FALSE, GetCurrentProcessId()); // open a handle to our own process
+		    if (PROCESS_HANDLE != INVALID_HANDLE_VALUE && PROCESS_HANDLE != NULL)
+		    {
+			    for (ULONG i = 0; i < handleTableInformation->NumberOfHandles; i++) 
+			    {	
+				    if ( (ULONG)handleTableInformation->Handles[i].UniqueProcessId == GetCurrentProcessId() ) 
+				    { 
+					    SYSTEM_HANDLE_TABLE_ENTRY_INFO handleInfo = (SYSTEM_HANDLE_TABLE_ENTRY_INFO)handleTableInformation->Handles[i];
+    					/* ROS comments shows ObTypeIdx as corrupted, would have helped with this process */
```

Comment:
Where exactly does it show it?

Reply:
I think while researching this I ran into a comment to that effect in a ReactOS source or header file. I cannot remember which file it was in (sorry). That note was a reminder for me to avoid using it and it should probably be removed..

Reply:
What are the changes requested for this PR ?  I could not find anything specific

Reply:
This sequence appears in ntoskrnl/ex/sysinfo.c and gave me the impression the type index data was not reliable:
#if 0 /* FIXME!!! Type field currupted */
                        HandleInformation->Handles[Index].ObjectTypeIndex =
                            (UCHAR) ObjectHeader->Type->Index;
#else
                        HandleInformation->Handles[Index].ObjectTypeIndex = 0;
#endif